ddtrace/contrib/aioredis/patch.py

```diff
@@ -0,0 +1,175 @@
+        return await func(*args, **kwargs)
+
+    decoded_args = [arg.decode() if isinstance(arg, bytes) else arg for arg in args]
+    with _trace_redis_cmd(pin, config.aioredis, instance, decoded_args):
+        return await func(*args, **kwargs)
+
+
+async def traced_pipeline(func, instance, args, kwargs):
+    pipeline = await func(*args, **kwargs)
+    pin = Pin.get_from(instance)
+    if pin:
+        pin.onto(pipeline)
+    return pipeline
+
+
+async def traced_execute_pipeline(func, instance, args, kwargs):
+    pin = Pin.get_from(instance)
+    if not pin or not pin.enabled():
+        return await func(*args, **kwargs)
+
+    cmds = [format_command_args(c) for c, _ in instance.command_stack]
+    resource = "\n".join(cmds)
+    with _trace_redis_execute_pipeline(pin, config.aioredis, resource, instance):
+        return await func(*args, **kwargs)
+
+
+def traced_13_pipeline(func, instance, args, kwargs):
+    pipeline = func(*args, **kwargs)
+    pin = Pin.get_from(instance)
+    if pin:
+        pin.onto(pipeline)
+    return pipeline
+
+
+def traced_13_execute_command(func, instance, args, kwargs):
+    # If we have a _RedisBuffer then we are in a pipeline
+    if isinstance(instance.connection, _RedisBuffer):
+        return func(*args, **kwargs)
+
+    pin = Pin.get_from(instance)
+    if not pin or not pin.enabled():
+        return func(*args, **kwargs)
+
+    decoded_args = [arg.decode() if isinstance(arg, bytes) else arg for arg in args]
+    # This span is unactive in the context because it should not have a child span
+    # Each consecutive execute command should create a sister span
+    span = pin.tracer.start_span(
+        redisx.CMD, service=trace_utils.ext_service(pin, config.aioredis), span_type=SpanTypes.REDIS, activate=False
+    )
+
+    span.set_tag(SPAN_MEASURED_KEY)
+    query = format_command_args(decoded_args)
+    span.resource = query
+    span.set_tag(redisx.RAWCMD, query)
+    if pin.tags:
+        span.set_tags(pin.tags)
+
+    span.set_tags(
+        {
+            net.TARGET_HOST: instance.address[0],
+            net.TARGET_PORT: instance.address[1],
+            redisx.DB: instance.db or 0,
+        }
+    )
+    span.set_metric(redisx.ARGS_LEN, len(decoded_args))
+    # set analytics sample rate if enabled
+    span.set_tag(ANALYTICS_SAMPLE_RATE_KEY, config.aioredis.get_analytics_sample_rate())
+
+    def _finish_span(future):
+        try:
+            # Accessing the result will raise an exception if:
+            #   - The future was cancelled
+            #   - There was an error executing the future (`future.exception()`)
+            #   - The future is in an invalid state
+            future.result()
+        except Exception:
+            span.set_exc_info(*sys.exc_info())
+        finally:
+            span.finish()
+
+    task = func(*args, **kwargs)
+    task.add_done_callback(_finish_span)
+    return task
+
+
+async def traced_13_execute_pipeline(func, instance, args, kwargs):
```

Comment:
we don't have network tags for this, e.g. `host`/`port`/`redis.db`.

I cannot remember why I wasn't able to get them or maybe I just forgot to add?

Might be worth taking a very quick look to see if we are able to get the underlying `_pool_or_conn` and get the tags from it, otherwise we should add a comment in the `with` block on why we cannot have those tags.